azurerm_mssql_virtual_machine resource causing panic crash

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and AzureRM Provider) Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

Terraform v0.12.24
AzureRM v2.35.0

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* `azurerm_mssql_virtual_machine`

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
module "base" {
  source  = "tfe.1corp.org/CCoE-Prod/windowsvmbase/azurerm"
  version = "~>3.0"

  workspace           = var.workspace
  region              = var.region
  name                = var.name
  size                = var.size
  resource_group_name = var.resource_group_name
  operating_system = {
    publisher = "MicrosoftSQLServer"
    offer     = "SQL2019-WS2019"
    sku       = local.environment == "d" ? "sqldev" : "enterprise"
    version   = "latest"
  }
  use_premium_storage_for_os = var.use_premium_storage_for_os
  data_disks                 = var.data_disks
  domain_join_credentials    = var.domain_join_credentials
  domain_join_ou             = var.domain_join_ou
  internet_access            = var.internet_access
  enable_automatic_updates   = true
}

resource "azurerm_mssql_virtual_machine" "this" {
  virtual_machine_id               = module.base.virtual_machine.id
  sql_license_type                 = local.environment == "d" ? "PAYG" : "AHUB"
  r_services_enabled               = true
  sql_connectivity_port            = 1433
  sql_connectivity_type            = "PRIVATE"
  sql_connectivity_update_username = module.base.credentials.user
  sql_connectivity_update_password = module.base.credentials.password

  auto_patching {
    day_of_week                            = "Sunday"
    maintenance_window_starting_hour       = 1
    maintenance_window_duration_in_minutes = 60 # must be in range 30 - 180
  }
}
```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

[https://gist.github.com/cub10s1ty/edaef4e9d16637b17ab4cc7630286921.js](url)

### Panic Output

I don't have access to the crash log in Terraform Enterprise, but I have the stack trace for the actual panic

```
panic: runtime error: invalid memory address or nil pointer dereference
2020-11-05T14:37:32.746Z [DEBUG] plugin.terraform-provider-azurerm_v2.35.0_x5: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x3ecc26f]
2020-11-05T14:37:32.746Z [DEBUG] plugin.terraform-provider-azurerm_v2.35.0_x5: 
2020-11-05T14:37:32.746Z [DEBUG] plugin.terraform-provider-azurerm_v2.35.0_x5: goroutine 896 [running]:
2020-11-05T14:37:32.746Z [DEBUG] plugin.terraform-provider-azurerm_v2.35.0_x5: github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/mssql.flattenArmSqlVirtualMachineAutoPatching(...)
2020-11-05T14:37:32.746Z [DEBUG] plugin.terraform-provider-azurerm_v2.35.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/mssql/mssql_virtual_machine_resource.go:383
2020-11-05T14:37:32.746Z [DEBUG] plugin.terraform-provider-azurerm_v2.35.0_x5: github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/mssql.resourceArmMsSqlVirtualMachineRead(0xc000e2ea10, 0x443e5c0, 0xc0014b8000, 0x0, 0x0)
2020-11-05T14:37:32.746Z [DEBUG] plugin.terraform-provider-azurerm_v2.35.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/mssql/mssql_virtual_machine_resource.go:312 +0x65f
2020-11-05T14:37:32.746Z [DEBUG] plugin.terraform-provider-azurerm_v2.35.0_x5: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000c72cf0, 0xc0000914a0, 0x443e5c0, 0xc0014b8000, 0xc000208598, 0x0, 0x0)
2020-11-05T14:37:32.746Z [DEBUG] plugin.terraform-provider-azurerm_v2.35.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource.go:460 +0x119
2020-11-05T14:37:32.746Z [DEBUG] plugin.terraform-provider-azurerm_v2.35.0_x5: github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ReadResource(0xc0001214d8, 0x53cfd00, 0xc00111cb70, 0xc00150b260, 0xc0001214d8, 0xc00111cb70, 0xc0019e4b78)
2020-11-05T14:37:32.746Z [DEBUG] plugin.terraform-provider-azurerm_v2.35.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin/grpc_provider.go:525 +0x3d8
2020-11-05T14:37:32.746Z [DEBUG] plugin.terraform-provider-azurerm_v2.35.0_x5: github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ReadResource_Handler(0x485ff00, 0xc0001214d8, 0x53cfd00, 0xc00111cb70, 0xc00150b200, 0x0, 0x53cfd00, 0xc00111cb70, 0xc001858700, 0x377)
2020-11-05T14:37:32.746Z [DEBUG] plugin.terraform-provider-azurerm_v2.35.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5/tfplugin5.pb.go:3269 +0x217
2020-11-05T14:37:32.746Z [DEBUG] plugin.terraform-provider-azurerm_v2.35.0_x5: google.golang.org/grpc.(*Server).processUnaryRPC(0xc000290900, 0x5411ae0, 0xc000291500, 0xc0015e4f00, 0xc000ac2870, 0x84fd8d0, 0x0, 0x0, 0x0)
2020-11-05T14:37:32.746Z [DEBUG] plugin.terraform-provider-azurerm_v2.35.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1024 +0x501
2020-11-05T14:37:32.747Z [DEBUG] plugin.terraform-provider-azurerm_v2.35.0_x5: google.golang.org/grpc.(*Server).handleStream(0xc000290900, 0x5411ae0, 0xc000291500, 0xc0015e4f00, 0x0)
2020-11-05T14:37:32.747Z [DEBUG] plugin.terraform-provider-azurerm_v2.35.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1313 +0xd3d
2020-11-05T14:37:32.747Z [DEBUG] plugin.terraform-provider-azurerm_v2.35.0_x5: google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc000bfc2f0, 0xc000290900, 0x5411ae0, 0xc000291500, 0xc0015e4f00)
2020-11-05T14:37:32.747Z [DEBUG] plugin.terraform-provider-azurerm_v2.35.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:722 +0xa1
2020-11-05T14:37:32.747Z [DEBUG] plugin.terraform-provider-azurerm_v2.35.0_x5: created by google.golang.org/grpc.(*Server).serveStreams.func1
2020-11-05T14:37:32.747Z [DEBUG] plugin.terraform-provider-azurerm_v2.35.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:720 +0xa1
```
<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->

### Expected Behavior

<!--- What should have happened? --->

### Actual Behavior

<!--- What actually happened? --->

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in a Azure China/Germany/Government? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Such as vendor documentation?
--->

--- a/azurerm/internal/services/mssql/mssql_virtual_machine_resource.go
+++ b/azurerm/internal/services/mssql/mssql_virtual_machine_resource.go
@@ -380,7 +380,7 @@ func expandArmSqlVirtualMachineAutoPatchingSettings(input []interface{}) *sqlvir
 }
 
 func flattenArmSqlVirtualMachineAutoPatching(autoPatching *sqlvirtualmachine.AutoPatchingSettings) []interface{} {
-	if autoPatching == nil || !*autoPatching.Enable {
+	if autoPatching == nil || autoPatching.Enable == nil || !*autoPatching.Enable {
 		return []interface{}{}
 	}
 

